packages/redux/src/raffles/actions/factories/createRaffleParticipationFactory.ts

```diff
@@ -0,0 +1,48 @@
+import * as actionTypes from '../../actionTypes';
+import {
+  Config,
+  PostParticipationData,
+  PostRaffleParticipation,
+  Raffle,
+  RaffleParticipation,
+  toBlackoutError,
+} from '@farfetch/blackout-client';
+import type { Dispatch } from 'redux';
+
+/**
+ * Method responsible to create a raffle participation.
+ *
+ * @param postRaffleParticipation - Create raffle participation.
+ *
+ * @returns Thunk factory.
+ */
+const createRaffleParticipationFactory =
+  (postRaffleParticipation: PostRaffleParticipation) =>
+  (raffleId: Raffle['id'], data: PostParticipationData, config?: Config) =>
+  async (dispatch: Dispatch): Promise<RaffleParticipation> => {
+    try {
+      dispatch({
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_REQUEST,
+      });
+      const participation = await postRaffleParticipation(
+        raffleId,
+        data,
+        config,
+      );
+
+      dispatch({
+        payload: { raffleId, participation },
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS,
+      });
+
+      return participation;
+    } catch (error) {
+      dispatch({
+        payload: { error: toBlackoutError(error) },
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE,
+      });
+
+      throw error;
```

Comment:
The error to be thrown should be the same that was dispatched on the payload of the failure action, so you need to create a temporary variable with the result of the `toBlackoutError` call and throw that as well as dispatching it through the failure action, like this:

```js
} catch (error) {
      const errorAsBlackoutError = toBlackoutError(error);

      dispatch({
        payload: { error: errorAsBlackoutError },
        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE,
      });

      throw errorAsBlackoutError;
```

Reply:
This is valid for all factories, please review them as well.

Reply:
done